File: src/main/scala/concurrency/tutorial/example_1_01.scala
package concurrency.tutorial

object example_1_01 extends App {

  object task extends Runnable {
    def run {
      val threadName = Thread.currentThread().getName()
      println("Hello " + threadName)
    }
  }

  task.run()

  val thread = new Thread(task)
  thread.start()

  println("Done!")

}

File: src/main/java/java8/concurrency/tutorial/Example_1_01.java
package java8.concurrency.tutorial;

public class Example_1_01 {

	public static void main(String[] args) {
		Runnable task = () -> {
			String threadName = Thread.currentThread().getName();
			System.out.println("Hello " + threadName);
		};

		task.run();

		Thread thread = new Thread(task);
		thread.start();

		System.out.println("Done!");
	}
}
